### PR TITLE
Add claims to support MP-JWT specification

### PR DESCRIPTION
**Pull request by madawas**
### Proposed changes in this pull request
- Added two new claims, upn and groups under oidc dialect in the `claim-config.xml` to support MP-JWT specification.

### When should this PR be merged

Need to evaluate whether mapping same local claim to multiple claims is correct or not and fix it if not correct.

#### General

- [x] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [x] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [x] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [x] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [x] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: features/claim-mgt/org.wso2.carbon.claim.mgt.server.feature/resources/conf/claim-config.xml [hunk 1]
```diff
@@ -1430,6 +1430,24 @@
 				<SupportedByDefault />
 				<MappedLocalClaim>http://wso2.org/claims/url</MappedLocalClaim>
 			</Claim>
+			<Claim>
+				<ClaimURI>upn</ClaimURI>
+				<DisplayName>User Principal</DisplayName>
+				<AttributeID>upn</AttributeID>
```
Comment (pulasthi7) [inline features/claim-mgt/org.wso2.carbon.claim.mgt.server.feature/resources/conf/claim-config.xml]: are these attributes available with the embedded LDAP by default?

Comment (emswbandara) [inline features/claim-mgt/org.wso2.carbon.claim.mgt.server.feature/resources/conf/claim-config.xml]: here AttributeID should be an available attribute in the default LDAP schema [1].
can't we map 'uid' as AttributeID here?

[1] https://github.com/wso2-extensions/identity-userstore-ldap/blob/master/features/org.wso2.carbon.ldap.server.server.feature/resources/conf/wso2Person.ldif

Comment (madawas) [reply features/claim-mgt/org.wso2.carbon.claim.mgt.server.feature/resources/conf/claim-config.xml]: @pulasthi7 Fixed with 1c6ddc3ba697f347c87ab31fc9bae90508fc3644 as @emswbandara suggested.

Comment (madawas) [reply features/claim-mgt/org.wso2.carbon.claim.mgt.server.feature/resources/conf/claim-config.xml]: @emswbandara Yes my mistake. It should map to `uid`. Thanks for pointing out. Fixed with 1c6ddc3ba697f347c87ab31fc9bae90508fc3644

**Pull request merged**